modules/auxiliary/scanner/misc/ibm_mq_enum.rb

```diff
@@ -0,0 +1,167 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Identify Queue Manager Name and MQ Version',
+      'Description'    => 'Run this auxiliary against the listening port of an IBM MQ Queue Manager to identify its name and version. Any channel type can be used to get this information as long as the name of the channel is valid.',
+      'Author'         => [ 'Petros Koutroumpis' ],
+      'License'        => MSF_LICENSE
+    ))
+    register_options(
+      [
+        OptString.new('CHANNEL', [ true, "Channel to use" ,"SYSTEM.DEF.SVRCONN"]),
+        OptInt.new('CONCURRENCY', [true, "The number of concurrent ports to check per host", 10]),
+        OptInt.new('TIMEOUT', [true, "The socket connect timeout in seconds", 5]),
+        OptString.new('PORTS', [true, 'Ports to probe', '1414']),
+
+      ])
+    deregister_options('RPORT')
+  end
+
+
+  def create_packet(channel_type)
+    chan = datastore['CHANNEL'] + "\x20"*(20-datastore['CHANNEL'].length.to_i)
+    if channel_type == 0
```

Comment:
It might be a good idea to future proof this method to handle and invalid specified `channel_type`.